src/PythHttpClient.ts

```diff
@@ -0,0 +1,138 @@
+import { AccountInfo, Commitment, Connection, PublicKey } from "@solana/web3.js";
+import { Product, PriceData, parseProductData, parsePriceData, parseBaseData, AccountType } from ".";
+
+const ONES = '1111111111111111111111111111111';
+
+interface PriceRawData {
+    key: PublicKey;
+    account: AccountInfo<Buffer>;
+}
+
+export interface PythHttpClientResult {
+    assetsTypes: string[];
+    productsSymbols: string[];
+    products: Product[];
+    productPrice: Map<string, PriceData>;
+    prices: PriceData[];
+}
+
+/**
+ * Reads Pyth price data from a solana web3 connection. This class uses a single HTTP call.
+ * Use the method getData() to get updated prices values.
+ */
+export class PythHttpClient {
+    connection: Connection;
+    pythProgramKey: PublicKey;
+    commitment: Commitment;
+
+    private productAccountKeyToProduct: Record<string, Product>;
+    private priceQueue: PriceRawData[];
+    private assetTypes: Set<string>;
+    private productSymbols: Set<string>;
+    private products: Set<Product>;
+    private productPrice: Map<string, PriceData>;
+    private prices: PriceData[];
+    public priceAccountKeyToProductAccountKey: Record<string, string>;
```

Comment:
why is this public?

Also, I think all of these variables could be local variables scoped to `handleAccount` right? I think that approach would be better because it doesn't have shared mutable state across invocations of that method. (if someone tried to call `handleAccount` in two parallel threads right now, something bad will happen)

Reply:
All attributes has been removed